tests/cartesian_tests/definitions.py

```diff
@@ -14,42 +14,47 @@
     cp = None
 
 import datetime
+from typing import Callable, List
 
 import numpy as np
 import pytest
 
 from gt4py import cartesian as gt4pyc
 from gt4py.cartesian import utils as gt_utils
-
-
-def _backend_name_as_param(name):
+from gt4py.cartesian.backend.base import Backend, from_name
+from gt4py.cartesian.testing.definitions import (
+    ALL_BACKEND_NAMES,
+    CPU_BACKEND_NAMES,
+    GPU_BACKEND_NAMES,
+    PERFORMANCE_BACKEND_NAMES,
+)
+from gt4py.cartesian.testing.suites import ParameterSet
+from gt4py.storage.cartesian.layout import is_gpu_device
+
+
+def _backend_name_as_param(backend_name: str) -> ParameterSet:
     marks = []
-    if gt4pyc.backend.from_name(name).storage_info["device"] == "gpu":
+    if backend_name in GPU_BACKEND_NAMES:
         marks.append(pytest.mark.requires_gpu)
-    if "dace" in name:
+    if "dace" in backend_name:
         marks.append(pytest.mark.requires_dace)
-    return pytest.param(name, marks=marks)
-
-
-_ALL_BACKEND_NAMES = list(gt4pyc.backend.REGISTRY.keys())
+    return pytest.param(backend_name, marks=marks)
 
 
-def _get_backends_with_storage_info(storage_info_kind: str):
+def _filter_backends(filter: Callable[[Backend], bool]) -> List[str]:
```

Comment:
Let's make sure the test generated are the same as before. `pytest --collect-only` will collect all combinations of tests and not run them